From c2004ef9b072c0116869272721ddbc45e33ab9f1 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 01 Apr 2020 19:22:41 +0300
Subject: [PATCH] [SW-11180] common: add __packed macro

Add the definition for __packed which is a short hand of
__attribute__((packed)).
U-Boot and Linux have this definition in their headers, but Preboot is
still does not.
With inclusion of the external hl_boot_desc.h header, we need to define
the __packed attribute, so it will get recognized.

Change-Id: I24423b357327b96ddd792c3694885d40f1e34fde
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/include/preboot.h b/include/preboot.h
index 33b04d2..28feb81 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -36,6 +36,8 @@
 
 #ifndef __ASSEMBLY__
 
+#define __packed	__attribute__((packed))
+
 typedef unsigned char u8;
 typedef unsigned int u32;
 typedef unsigned long long u64;
